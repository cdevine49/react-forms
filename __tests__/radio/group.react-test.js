import React from 'react';
import RadioGroup from '../../src/components/radio/group';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('RadioGroup', () => {
  test('Default', () => {
    const tree = renderer.create(
      <RadioGroup
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('Fieldset', () => {
    test('accepts props by default', () => {
      const tree = renderer.create(
        <RadioGroup
          id='field-radio'
          className='add-me'
          data-test='some data'
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

  });

  describe('Legend', () => {
    test('doesn\'t render without legend prop', () => {
      const tree = renderer.create(
        <RadioGroup
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('accepts legend prop as child', () => {
      const tree = renderer.create(
        <RadioGroup
          legend="Keep this short"
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('accepts legendProps object as props', () => {
      const tree = renderer.create(
        <RadioGroup
          legend="Legend name"
          legendProps={{
            id: 'legend-id',
            className: 'extra classes'
          }}
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Children', () => {
    const child = ({ props }) => <div { ...props }></div>;

    test('default props', () => {
      const tree = renderer.create(
        <RadioGroup>
          <child />
          <child />
          <child />
        </RadioGroup>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
    test('Child with value prop equal to group\'s currentValue prop is checked', () => {
      let tree = renderer.create(
        <RadioGroup currentValue={1}>
          <child value={0} />
          <child value={1} />
          <child value={2} />
        </RadioGroup>
      ).toJSON();

      expect(tree).toMatchSnapshot();

      tree = renderer.create(
        <RadioGroup currentValue={2}>
          <child value={0} />
          <child value={1} />
          <child value={2} />
        </RadioGroup>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    describe('OnChange', () => {
      const onChange = jest.fn();
      test('received by each child', () => {
        const tree = renderer.create(
          <RadioGroup onChange={onChange}>
            <child />
            <child />
            <child />
          </RadioGroup>
        ).toJSON();

        expect(tree).toMatchSnapshot();
      });

      const radioGroup = shallow(
        <RadioGroup onChange={onChange}>
          <child className="child" />
          <child className="child" />
          <child className="child" />
        </RadioGroup>
      );
      let calledCount = 0;
      radioGroup.find('.child').forEach(child => {
        test('can be called by each child', () => {
          expect(onChange).toHaveBeenCalledTimes(calledCount);
          child.props().onChange();
          calledCount++;
          expect(onChange).toHaveBeenCalledTimes(calledCount);
        });
      });
    });
  });

  describe('Error', () => {
    test('doesn\'t render without errorMessage prop', () => {
      const tree = renderer.create(
        <RadioGroup
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('accepts errorMessage prop as child', () => {
      const tree = renderer.create(
        <RadioGroup
          errorMessage="Whoopsie"
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('accepts errorProps object as props', () => {
      const tree = renderer.create(
        <RadioGroup
          errorMessage="Uh Oh"
          errorProps={{
            id: 'ya-dun-goofed',
            className: 'way to go buddy'
          }}
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Underline', () => {
    test('doesn\'t render without underline prop', () => {
      const tree = renderer.create(
        <RadioGroup
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('accepts underline prop as child', () => {
      const tree = renderer.create(
        <RadioGroup
          underline="A bit of extra explanation"
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test('accepts underlineProps object as props', () => {
      const tree = renderer.create(
        <RadioGroup
          underline="More info"
          underlineProps={{
            id: 'what-is-going-on',
            className: 'i am here to help'
          }}
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});