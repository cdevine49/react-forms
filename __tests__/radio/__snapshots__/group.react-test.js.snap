// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup Children Child with value prop equal to group's currentValue prop is checked 1`] = `
<fieldset
  className="radio-button-container"
>
  <child
    checked={false}
    onChange={[Function]}
    onKeyPress={[Function]}
    value={0}
  />
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
    value={1}
  />
  <child
    checked={false}
    onChange={[Function]}
    onKeyPress={[Function]}
    value={2}
  />
</fieldset>
`;

exports[`RadioGroup Children Child with value prop equal to group's currentValue prop is checked 2`] = `
<fieldset
  className="radio-button-container"
>
  <child
    checked={false}
    onChange={[Function]}
    onKeyPress={[Function]}
    value={0}
  />
  <child
    checked={false}
    onChange={[Function]}
    onKeyPress={[Function]}
    value={1}
  />
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
    value={2}
  />
</fieldset>
`;

exports[`RadioGroup Children OnChange received by each child 1`] = `
<fieldset
  className="radio-button-container"
>
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
</fieldset>
`;

exports[`RadioGroup Children OnKeyPress received by each child 1`] = `
<fieldset
  className="radio-button-container"
>
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
</fieldset>
`;

exports[`RadioGroup Children default props 1`] = `
<fieldset
  className="radio-button-container"
>
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
  <child
    checked={true}
    onChange={[Function]}
    onKeyPress={[Function]}
  />
</fieldset>
`;

exports[`RadioGroup Default 1`] = `
<fieldset
  className="radio-button-container"
/>
`;

exports[`RadioGroup Error accepts errorMessage prop as child 1`] = `
<fieldset
  className="radio-button-container"
>
  <p
    className="error-message radio-group-error"
  >
    Whoopsie
  </p>
</fieldset>
`;

exports[`RadioGroup Error accepts errorProps object as props 1`] = `
<fieldset
  className="radio-button-container"
>
  <p
    className="error-message way to go buddy"
    id="ya-dun-goofed"
  >
    Uh Oh
  </p>
</fieldset>
`;

exports[`RadioGroup Error doesn't render without errorMessage prop 1`] = `
<fieldset
  className="radio-button-container"
/>
`;

exports[`RadioGroup Fieldset accepts props by default 1`] = `
<fieldset
  className="radio-button-container add-me"
  data-test="some data"
  id="field-radio"
/>
`;

exports[`RadioGroup Legend accepts legend prop as child 1`] = `
<fieldset
  className="radio-button-container"
>
  <legend
    className="radio-group-legend"
  >
    Keep this short
  </legend>
</fieldset>
`;

exports[`RadioGroup Legend accepts legendProps object as props 1`] = `
<fieldset
  className="radio-button-container"
>
  <legend
    className="radio-group-legend extra classes"
    id="legend-id"
  >
    Legend name
  </legend>
</fieldset>
`;

exports[`RadioGroup Legend doesn't render without legend prop 1`] = `
<fieldset
  className="radio-button-container"
/>
`;

exports[`RadioGroup Underline accepts underline prop as child 1`] = `
<fieldset
  className="radio-button-container"
>
  <p
    className="underline radio-group-underline"
  >
    A bit of extra explanation
  </p>
</fieldset>
`;

exports[`RadioGroup Underline accepts underlineProps object as props 1`] = `
<fieldset
  className="radio-button-container"
>
  <p
    className="underline i am here to help"
    id="what-is-going-on"
  >
    More info
  </p>
</fieldset>
`;

exports[`RadioGroup Underline doesn't render without underline prop 1`] = `
<fieldset
  className="radio-button-container"
/>
`;
